Libraries/LibURL/Host.cpp

```diff
@@ -225,15 +225,30 @@ Optional<String> Host::registrable_domain() const
     auto trailing_dot = host_string.ends_with('.') ? "."sv : ""sv;
 
     // 3. Let registrableDomain be the registrable domain determined by running the Public Suffix List algorithm with host as domain. [PSL]
-    // FIXME: This is not correct, we should be doing the same as public_suffix() above.
-    auto registrable_domain = get_registrable_domain(host_string).value_or("*"_string);
+    //
+    // NOTE: If we do not find a registrable domain via the PSL, use everything after the second to last dot.
+    auto registrable_domain = get_registrable_domain(host_string);
+    if (!registrable_domain.has_value()) {
+        auto view = host_string.bytes_as_string_view();
+
+        auto last_dot = view.find_last('.');
+        if (last_dot.has_value()) {
+            view = view.substring_view(0, *last_dot);
+            auto second_last_dot = view.find_last('.');
+            if (second_last_dot.has_value())
+                registrable_domain = MUST(host_string.substring_from_byte_offset(second_last_dot.value() + 1));
+        }
+    }
+
+    if (!registrable_domain.has_value())
+        registrable_domain = host_string;
```

Comment:
Commit message: two things, the title states "in case of missing in PSL" which seems to be missing a word. Maybe this would work better?

> LibURL: Return a proper registrable domain if it's missing in the PSL

Also I find this hard to read:

> This fixes us, for example, considering https://b.b.example/ and https://a.example/ as same site to one another.

Suggestion:

> This prevents us from considering e.g. https://b.b.example/ and https://a.example/ as the same site.